src/data/reports/glass-data-submission/GLASSDataSubmissionDefaultRepository.ts

```diff
@@ -154,13 +168,77 @@ export class GLASSDataSubmissionDefaultRepository implements GLASSDataSubmission
             };
         });
 
-        const filteredRows = rows.filter(
-            row =>
-                (_.isEmpty(orgUnitIds) || !row.orgUnit ? row : orgUnitIds.includes(row.orgUnit)) &&
-                periods.includes(String(row.period)) &&
-                (completionStatus !== undefined ? row.questionnaireCompleted === completionStatus : row) &&
-                (!submissionStatus ? row : row.status === submissionStatus)
+        const quarterPeriods = _.flatMap(periods, year => quarters.map(quarter => `${year}${quarter}`));
+
+        const filteredRows = rows.filter(row => {
+            return (_.isEmpty(orgUnitIds) || !row.orgUnit ? row : orgUnitIds.includes(row.orgUnit)) &&
```

Comment:
These conditionals are very difficult to understand all together. Perhaps to make them easier to understand and to avoid future errors if something needs to be changed, they could be created in variables with descriptive names